src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -193,7 +200,8 @@ SnapTrimEvent::start()
         });
       });
     });
-  }, [this](std::exception_ptr eptr) -> snap_trim_ertr::future<seastar::stop_iteration> {
+  }, [this, ref]
```

Comment:
Does the issue disappear after this commit?

But my understanding is that capturing `ref` in the `finally()` continuation below should make it unnecessary to capture `ref` in the middle continuations from this commit.

Reply:
The backtrace in this commit (abceb1652239629ed11187a5fc670a3b1a3a5bb1) occurred based on the commit that captured `ref` in the `finaly()` continuation below (84c5b6c578c1e64807eca5098d51193ee71e6106).

See osd.3: https://pulpito.ceph.com/matan-2023-11-16_08:02:56-crimson-rados-wip-matanb-crimson-testing-11-15-distro-crimson-smithi/7460420/

Build: https://github.com/ceph/ceph-ci/commits/wip-matanb-crimson-testing-11-15
Commit: a72569d5def44c563b4d6b860112880e1e08b106

> Does the issue disappear after this commit?

According to the last round, yes.

Reply:
IIUC, the capture here should be different that usual because we ignore the future returned.
```
void PG::on_active_actmap()
{
  std::ignore = seastar::do_until(
    [this] {...},
    [this] {
      return seastar::repeat([to_trim, needs_pause, this] {
        return shard_services.start_operation<SnapTrimEvent>(...)

Reply:
I'm still wondering that capturing `ref` in `finally()` (L207) should extend its life until `finally()` is called, which implies that capturing in the previous chained continuations (L100, L204) before `finally()` are unnecessary.

(IIUC the std::ignore above is unrelated to this internal chaining.)

> According to the last round, yes.

I'm still feeling skeptical here, but anyway, it'd be fine to go ahead and watch the future tests.

Reply:
> I'm still feeling skeptical here, but anyway, it'd be fine to go ahead and watch the future tests.

I run 20 thrash test (with snapshot activity) which had the crashes mentioned here, it seems that we no longer crash on these test (19/20, 1 unrelated).
https://pulpito.ceph.com/matan-2023-11-20_10:00:00-crimson-rados-wip-matanb-crimson-testing-11-19-distro-crimson-smithi/